spec/attribute.dd

```diff
@@ -617,27 +617,117 @@ auto i = 6.8;   // declare i as a double
 $(H2 $(LNAME2 scope, $(D scope) Attribute))
 
 $(P
-        The $(D scope) attribute is used for local variables and implements the
-        RAII (Resource Acquisition Is Initialization) protocol.
+        The $(D scope) attribute is used for parameters or local variables with a type that contains low-level pointers.
+        It signifies that the variable's pointer values will not escape the function
```

Comment:
This section is clumsily worded, I'll be more specific when I'm at a proper keyboard.

Reply:
I suggest using smaller sentences. Also, it should be clear that any variable may be attributed with `scope`, however, the checks are performed only for types that contain indirections. The way this paragraph is worded makes it sound that it is an error to use scope on types that do not contain indirections.

Reply:
For example: "The scope attribute is used for parameters of local variables. If the type of the variable does not contain any indirections, scope is ignored. Otherwise, the compiler will mechanically check that the variable does not escape the scope it is declared in. A variable may escape its scope by being returned or being assigned to a variable outside of its scope".